setuptools/_normalization.py

```diff
@@ -12,7 +12,7 @@
 _VALID_NAME = re.compile(r"^([A-Z0-9]|[A-Z0-9][A-Z0-9._-]*[A-Z0-9])$", re.I)
 _UNSAFE_NAME_CHARS = re.compile(r"[^A-Z0-9._-]+", re.I)
 _NON_ALPHANUMERIC = re.compile(r"[^A-Z0-9]+", re.I)
-_PEP440_FALLBACK = re.compile(r"^v?(?P<safe>(?:[0-9]+!)?[0-9]+(?:\.[0-9]+)*)", re.I)
+_PEP440_FALLBACK = re.compile(r"^(?P<safe>v?(?:[0-9]+!)?[0-9]+(?:\.[0-9]+)*)", re.I)
```

Comment:
This is an existing bug that I have identified when working on this PR.
I added a new doctest to avoid regression down in the same file:

```python
    >>> best_effort_version("v78.1.0-2-g3a3144f0d")
    '78.1.0.dev0+sanitized.2.g3a3144f0d'
```